src/main/java/com/appdynamics/monitors/azure/AzureMonitorTask.java

```diff
@@ -8,120 +8,99 @@
+        else {
+            subscriptionName = subscription.get("subscriptionId").toString();
         }
-        if (logger.isDebugEnabled()) {
-            assert url != null;
-            logger.debug("Get Metrics REST API Request: " + url.toString());}
-        assert url != null;
-        extractMetrics(AzureRestOperation.doGet(azureAuth,url));
-    }
 
-    private void extractMetrics(JsonNode json){
-        if (logger.isDebugEnabled()) {logger.debug("Get Metrics Response JSON: " + Utilities.prettifyJson(json));}
-        JsonNode jsonValue = json.get("value");
-        for (JsonNode currentValueNode:jsonValue){
-            String metricId = extractMetridId(currentValueNode.get("id").asText());
-            String metricNameValue = currentValueNode.get("name").get("value").asText();
-            String metricUnit = currentValueNode.get("unit").asText();
-            String metricType = null;
-            BigDecimal metricValue = null;
-            if(currentValueNode.get("timeseries").has(0)){
-                JsonNode jsonData = currentValueNode.get("timeseries").get(0).get("data");
-                for (JsonNode currentDataNode:jsonData){
-                    if (currentDataNode.has("average")){
-                        metricType = "average"; metricValue = currentDataNode.get("average").decimalValue();
-                    }
-                    else if (currentDataNode.has("total")){
-                        metricType = "total"; metricValue = currentDataNode.get("total").decimalValue();
-                    }
-                    else if (currentDataNode.has("last")){
-                        metricType = "last"; metricValue = currentDataNode.get("last").decimalValue();
-                    }
-                    else if (currentDataNode.has("maximum")){
-                        metricType = "maximum"; metricValue = currentDataNode.get("maximum").decimalValue();
+        for (Map<String, ?> resourceGroupFilter : resourceGroupFilters) {
+            List<Map<String, ?>> resourceTypeFilters = (List<Map<String, ?>>) resourceGroupFilter.get("resourceTypes");
+            for (Map<String, ?> resourceTypeFilter : resourceTypeFilters) {
+                List<Map<String, ?>> resourceFilters = (List<Map<String, ?>>) resourceTypeFilter.get("resources");
+                for (Map<String, ?> resourceFilter : resourceFilters) {
+                    String currentResourceGroupFilter = resourceGroupFilter.get("resourceGroup").toString();
+                    for (GenericResource resource : resources) {
+                        String currentResourceFilter = resourceFilter.get("resource").toString();
+                        String currentResourceTypeFilter = resourceTypeFilter.get("resourceType").toString();
+                        AzureMetrics azureMetricsTask = new AzureMetrics(
+                                resourceFilter,
+                                currentResourceGroupFilter,
```

Comment:
This is a very expensive operation (n ^ 4). I understand why it is that way, looking at the current hierarchy in your config.yml but I'm wondering if this can be done more efficiently. Any thoughts?